src/ashell/webusb_driver.c

```diff
@@ -0,0 +1,430 @@
+    LOW_BYTE(WEBUSB_VENDOR_ID),
+    HIGH_BYTE(WEBUSB_VENDOR_ID),       /* Vendor Id */
+    LOW_BYTE(WEBUSB_PRODUCT_ID),
+    HIGH_BYTE(WEBUSB_PRODUCT_ID),      /* Product Id */
+    LOW_BYTE(BCDDEVICE_RELNUM),
+    HIGH_BYTE(BCDDEVICE_RELNUM),    /* Device Release Number */
+    /* Index of Manufacturer String Descriptor */
+    0x01,
+    /* Index of Product String Descriptor */
+    0x02,
+    /* Index of Serial Number String Descriptor */
+    0x03,
+    WEBUSB_NUM_CONF,                /* Number of Possible Configuration */
+
+    /* Configuration descriptor */
+    USB_CONFIGURATION_DESC_SIZE,    /* Descriptor size */
+    USB_CONFIGURATION_DESC,         /* Descriptor type */
+    /* Total length in bytes of data returned */
+    LOW_BYTE(WEBUSB_SERIAL_CONF_SIZE),
+    HIGH_BYTE(WEBUSB_SERIAL_CONF_SIZE),
+    WEBUSB_NUM_ITF,                 /* Number of interfaces */
+    0x01,                           /* Configuration value */
+    0x00,                           /* Index of the Configuration string */
+    USB_CONFIGURATION_ATTRIBUTES,   /* Attributes */
+    MAX_LOW_POWER,                  /* Max power consumption */
+
+    /* Interface descriptor */
+    USB_INTERFACE_DESC_SIZE,        /* Descriptor size */
+    USB_INTERFACE_DESC,             /* Descriptor type */
+    0x00,                           /* Interface index */
+    0x00,                           /* Alternate setting */
+    WEBUSB_NUM_EP,                  /* Number of Endpoints */
+    CUSTOM_CLASS,                   /* Custom Class */
+    0x00,                           /* SubClass */
+    0x00,                           /* Protocol */
+    /* Index of the Interface String Descriptor */
+    0x00,                           /* Interval */
+
+    /* First Endpoint IN */
+    USB_ENDPOINT_DESC_SIZE,         /* Descriptor size */
+    USB_ENDPOINT_DESC,              /* Descriptor type */
+    WEBUSB_ENDP_IN,                 /* Endpoint address */
+    USB_DC_EP_BULK,                 /* Attributes */
+    LOW_BYTE(WEBUSB_BULK_EP_MPS),
+    HIGH_BYTE(WEBUSB_BULK_EP_MPS),  /* Max packet size */
+    0x00,                           /* Interval */
+
+    /* Second Endpoint OUT */
+    USB_ENDPOINT_DESC_SIZE,         /* Descriptor size */
+    USB_ENDPOINT_DESC,              /* Descriptor type */
+    WEBUSB_ENDP_OUT,                /* Endpoint address */
+    USB_DC_EP_BULK,                 /* Attributes */
+    LOW_BYTE(WEBUSB_BULK_EP_MPS),
+    HIGH_BYTE(WEBUSB_BULK_EP_MPS),  /* Max packet size */
+    0x00,                           /* Interval */
+
+    /* String descriptor language, only one, so min size 4 bytes.
+     * 0x0409 English(US) language code used
+     */
+    USB_STRING_DESC_SIZE,           /* Descriptor size */
+    USB_STRING_DESC,                /* Descriptor type */
+    0x09,
+    0x04,
+
+    /* Manufacturer String Descriptor "Intel" */
+    0x0C,
+    USB_STRING_DESC,
+    'I', 0, 'n', 0, 't', 0, 'e', 0, 'l', 0,
```

Comment:
It would sure be nice as you look to upstreaming this, if you could make these strings configurable... I know it's not that way in the current driver either.

Reply:
Thanks, point taken.